### PR TITLE
Drop iOS 14 Support: Increase deployment target to iOS 15

### PR DESCRIPTION
**Pull request by kean**
Related Issue #20860

Bump the deployment target to iOS 15 and remove the now redundant #available checks and the associated code.

To test: make sure it still compiles and runs (run some basic smoke tests)

## Regression Notes
1. Potential unintended areas of impact: n/a
2. What I did to test those areas of impact (or what existing automated tests I relied on): n/a
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by hassaanelgarem — APPROVED** [review]
Thanks for wrangling this @kean!

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift
Comment (guarani) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift:341]: This is the only change I'm not sure of. It looks like it's the Quick Start popover but I'm not sure how it's relevant to dropping iOS 14. 
Do you have more insight into this, @kean?

Comment (kean) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift:341]: The only code in Dashboard that I found that made use of this delegate is also removed as part of this PR:

```swift
// Fallback context menu implementation for iOS 13.
func showMenuSheet() {
    // ...
   
    if let popoverPresentationController = menuViewController.popoverPresentationController {
            popoverPresentationController.delegate = presenterViewController

   // ...
}
```

You have to set the delegate in order to it to get called, so it seems it's safe to remove.

I'm planning to remove [`MenuSheetViewController`](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Classes/ViewRelated/Views/MenuSheetViewController.swift) as well, but there is one use left in `ReaderCommentsViewController`, which I'm not sure is safe to remove – I'll investigate.

Comment (guarani) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift:341]: > You have to set the delegate in order to it to get called, so it seems it's safe to remove.

Gotcha 👍 



> I'm planning to remove [`MenuSheetViewController`](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Classes/ViewRelated/Views/MenuSheetViewController.swift) as well, but there is one use left in `ReaderCommentsViewController`, which I'm not sure is safe to remove – I'll investigate.

Thanks for taking a look.

I'll approve now since this PR looks good!

**Review by guarani — COMMENTED** [review]
I noticed the test plan only calls for the CI to be green. Do you think smoke testing of the build should be done? I'm not sure what testing has already been done. 

With the next release being cut in on Monday (see PaaHJt-eZ-p2), I'm wondering if there's enough time this week for any issues related to the deployment target bump to be caught on the `trunk` branch.

**Comment by kean** [comment]
> I noticed the test plan only calls for the CI to be green. Do you think smoke testing of the build should be done? I'm not sure what testing has already been done.

I've been testing previous PRs individually, and I've also did quick regression testing yesterday.

With this PR, it's hard to pinpoint what areas to test, but I at least tested the main areas of the app.

> I'm wondering if there's enough time this week for any issues related to the deployment target bump to be caught on the trunk branch.

I think it should be OK, but we could wait until Tuesday to merge it. There is no rush. But most of the related changes were already merged.

**Review by guarani — APPROVED** [review]
> I've been testing previous PRs individually, and I've also did quick regression testing yesterday.
> With this PR, it's hard to pinpoint what areas to test, but I at least tested the main areas of the app.

Ok, I did a quick tap through the JP and WP app builds to try to catch regressions and had none.

> I think it should be OK, but we could wait until Tuesday to merge it. There is no rush. But most of the related changes were already merged.

Ok, if we merge today there's only today and tomorrow to catch issues before the next beta (22.7). However, I don't see a big risk here as I've gone through the code changes and nothing looks out of place.

As for the actual deployment target change from 14 to 15, I can't think of any bugs that could come of that.

So, feel free to merge this week or next 👍 

Thanks for your work, @kean!

**Comment by mokagio** [comment]
> Ok, if we merge today there's only today and tomorrow to catch issues before the next beta (22.7). However, I don't see a big risk here as I've gone through the code changes and nothing looks out of place.

On top of that, I would argue that only major issues would be noticed because we don't distribute nightly builds from `trunk` and don't have a mechanism for a large enough pool of folks using the app not smoke testing it to get their hands on these changes.

**Comment by kean** [comment]
Hey, @Guarani. The [release branch](https://github.com/wordpress-mobile/WordPress-iOS/tree/release/22.7) was cut, so I'm planning to merge it today, and then I'll continue working on the remaining deprecations when I have time. I'm prioritizing the work on Blaze.

**Pull request merged**
